src/Renderer/NavigatorRenderer.cpp

```diff
@@ -0,0 +1,378 @@
+
+  TextRenderer text_renderer2;
+  text_renderer2.SetVCenter(true);
+  text_renderer2.SetCenter(true);
+  text_renderer2.SetControl(true);
+  
+  canvas.Select(*look.font);
+
+  bool has_started = calculated.ordered_task_stats.start.task_started;
+
+  const auto time_elapsed = TimeStamp{FloatDuration{calculated.ordered_task_stats.total.time_elapsed}};
+  const RoughTimeDelta r{};
+
+  auto time_elapsed_s = FormatLocalTimeHHMM(time_elapsed, r).c_str();
+  has_started ? time_elapsed_s : time_elapsed_s = "--:--";
+
+  const auto time_start = calculated.ordered_task_stats.start.time;
+  auto time_start_s = FormatLocalTimeHHMM(time_start,
+                                  CommonInterface::GetComputerSettings().utc_offset).c_str();
+  has_started ? time_start_s : time_start_s = "--:--";
+
+  const auto time_local = FormatLocalTimeHHMM(basic.time,
+                                  CommonInterface::GetComputerSettings().utc_offset).c_str();
+
+  const auto time_planned = TimeStamp{FloatDuration{calculated.ordered_task_stats.total.time_planned}};
+  auto time_planned_s = FormatLocalTimeHHMM(time_planned, r).c_str();
+  has_started ? time_planned_s : time_planned_s = "--:--";
+
+  const auto arrival_planned = TimeStamp{FloatDuration{time_start.ToDuration() + time_planned.ToDuration()}};
+  auto arrival_planned_s = FormatLocalTimeHHMM(arrival_planned, 
+                                  CommonInterface::GetComputerSettings().utc_offset).c_str(); 
+  has_started ? arrival_planned_s : arrival_planned_s = "--:--";
+
+  auto waypoint_s = static_cast<std::string>(wp_current.name);
+
+  // WP Dist
+  const auto waypoint_distance = calculated.ordered_task_stats.current_leg.vector_remaining.distance;
+
+  std::ostringstream oss;
+  oss << std::fixed << std::setprecision(2) << std::noshowpoint  << waypoint_distance/1000; 
+  auto waypoint_distance_s = oss.str(); 
+
+  // WP ALTD
+  const auto waypoint_altitude_diff = static_cast<int>(calculated.ordered_task_stats.current_leg.solution_remaining.GetRequiredAltitude());
+  auto waypoint_altitude_diff_s = std::to_string(waypoint_altitude_diff); 
+
+  // WP GR
+  const auto waypoint_GR = calculated.ordered_task_stats.current_leg.gradient;
+  auto waypoint_GR_s = std::to_string(static_cast<int>(waypoint_GR));
+
+  // e_WP_BearingDiff
+  Angle bearing_diff{};
+  bearing_diff.Zero();
+  if (basic.track_available) 
+    bearing_diff = calculated.ordered_task_stats.current_leg.vector_remaining.bearing - basic.track;
+  auto waypoint_direction = static_cast<int>(bearing_diff.AsDelta().Degrees());
+  auto waypoint_direction_s = std::to_string(waypoint_direction);
+
+  const auto caption1 = waypoint_distance_s + " km | " + waypoint_altitude_diff_s + " m | " + waypoint_GR_s + ":1";
+            // + _T("\t\t\t\t--- ") + current_speed_s + " | " + waypoint_direction_s + " ---\n";
+
+  const auto caption2 = std::string(time_start_s) \
+            + _T("\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t     ") \
```

Comment:
Please no on-screen layout using tabs...

Reply:
OK, instead of concatenating all texts together, I cut this caption2 in 3 parts and placed them according to the width.
tabs deleted